### ISSUE TITLE
Build an Async python client

### ISSUE THREAD

**Issue by iJanki-gr** [armadaproject/armada#1992]
To improve Airflow scalability a useful technique is to use a deferred operator.

https://airflow.apache.org/docs/apache-airflow/stable/concepts/deferring.html

"Standard [Operators](https://airflow.apache.org/docs/apache-airflow/stable/concepts/operators.html) and [Sensors](https://airflow.apache.org/docs/apache-airflow/stable/concepts/sensors.html) take up a full worker slot for the entire time they are running, even if they are idle; for example, if you only have 100 worker slots available to run Tasks, and you have 100 DAGs waiting on a Sensor that’s currently running but idle, then you cannot run anything else - even though your entire Airflow cluster is essentially idle. reschedule mode for Sensors solves some of this, allowing Sensors to only run at fixed intervals, but it is inflexible and only allows using time as the reason to resume, not anything else.

This is where Deferrable Operators come in. A deferrable operator is one that is written with the ability to suspend itself and free up the worker when it knows it has to wait, and hand off the job of resuming it to something called a Trigger. As a result, while it is suspended (deferred), it is not taking up a worker slot and your cluster will have a lot less resources wasted on idle Operators or Sensors."

An example of this is the standard KubernetesPodOperator at
https://github.com/apache/airflow/blob/main/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L129

and its async counterpart at:
https://github.com/astronomer/astronomer-providers/blob/main/astronomer/providers/cncf/kubernetes/operators/kubernetes_pod.py#L22

They can practically be swapped for one another in code.

Can we start a conversation to see if it makes sense to build an async version of armadaoperator too?

┆Issue is synchronized with this [Jira Task](https://gr-oss.atlassian.net/browse/BATCH-100) by [Unito](https://www.unito.io)

**Comment by kannon92** [comment]
Thank you for posting this issue! Our team is fully focused on another project at the moment. We have been discussing this and it’s very interesting to us! 

A few of us will be in London from Feb 6th to 10th so that could be a good time to discuss this and the airflow project as a whole.

**Comment by Chinwendu20** [comment]
I would like to work on this please

**Comment by dave-gantenbein** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Armada-62c744dc3b819100132ca032/issues/armadaproject/armada/1992?planning-poker) with Zenhub @dejanzele @kannon92 @richscott @Sharpz7 @suprjinx

**Comment by dave-gantenbein** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Armada-62c744dc3b819100132ca032/issues/armadaproject/armada/1992?planning-poker) with Zenhub @ClifHouck @dejanzele @headphonejames @kannon92 @richscott @Sharpz7 @suprjinx
